### ISSUE TITLE
Random walk phase estimation example in Python

### ISSUE THREAD

**Issue by bettinaheim** [NVIDIA/cuda-quantum#1579]
CUDA-Q has a random walk phase estimation example implemented in C++. With the new Python support added in 0.7.0 it should now be possible to also express this kind of real-time processing in Python. It would be nice to add the corresponding Python example.

**Comment by Gaurang-Belekar** [comment]
Hi @bettinaheim, I have worked on the random walk phase estimation algorithm. If you can assign the issue to me, I will would be happy to contribute the python code. To be clear, the issue you mentioned is referring to the Iterative Phase Estimation ([link](https://nvidia.github.io/cuda-quantum/latest/specification/cudaq/examples.html#iterative-phase-estimation)) as show in the cuda Quantum document.

**Comment by bettinaheim** [comment]
Hi @Gaurang-Belekar, 
Sorry I missed your comment. No, the issue refers to the [random walk phase estimation](https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/examples/cpp/other/random_walk_qpe.cpp). Let me know if you still want to pick it up.

**Comment by Gaurang-Belekar** [comment]
Hi @bettinaheim ,
I would still like to work on this as well. Again to be clear there is a need of a python version of the random walk phase estimation cpp file as mentioned [above](https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/examples/cpp/other/random_walk_qpe.cpp).

**Comment by bettinaheim** [comment]
> Hi @bettinaheim , I would still like to work on this as well. Again to be clear there is a need of a python version of the random walk phase estimation cpp file as mentioned [above](https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/examples/cpp/other/random_walk_qpe.cpp).

Correct. If you want to give it a go, please fork the repo and create a PR for it. Let us know if you run into any trouble. :)

**Comment by Gaurang-Belekar** [comment]
Hi @bettinaheim , I have created the python version of the random walk qpe. Tested it as well and gives similar result. Please review and let me know if there are any changes needed. 

Also if there are any other PR I work with, I am happy to help :)

Thanks.

**Comment by khalatepradnya** [comment]
Completed by https://github.com/NVIDIA/cuda-quantum/pull/2060.

See: https://github.com/NVIDIA/cuda-quantum/blob/f662540b96e2546b014b1aa729b093565593233e/docs/sphinx/examples/python/random_walk_qpe.py
